### PR TITLE
Add new endpoint /metrics with relevant system monitoring metrics

### PR DESCRIPTION
**Pull request by rosswhitfield**
# Description of the changes

Ref [7060: [WebMon] Add new endpoint /metrics with relevant system monitoring metrics](https://ornlrse.clm.ibmcloud.com/ccm/resource/itemName/com.ibm.team.workitem.WorkItem/7060)

This add new "metrics" app that will return data in `json` useful for monitoring. The is an endpoint that returns all the data or you can request specific data. All timestamps will be ISO8601.

To test, start up webmon with docker compose.

Then after logging end you can visit

* http://localhost/metrics/ (for all the data at once)
* http://localhost/metrics/workflow_diagnostics/ (for workflow diagnostics)
* http://localhost/metrics/postprocessing_diagnostics/ (for postprocessing diagnostics)
* http://localhost/metrics/instrument_status/ (for instrument status)
* http://localhost/metrics/run_statuses/ (return the count of the current reduction statuses for all the run started in the last hour)
* http://localhost/metrics/run_statuses/{n}/ (return the count of the current reduction statuses for all the run started in the n minutes, _e.g._ http://localhost/metrics/run_statuses/525600/ will return the last year)

Should look something like
2024-09-26-160301_500x710_scrot

Anything else I should include?

Check all that apply:
- [ ] updated documentation
- [ ] Source added/refactored
- [ ] Added unit tests
- [ ] Added integration tests
- [ ] (If applicable) Verified that manual tests requiring the /SNS and /HFIR filesystems pass without fail

**References:**
- Links to IBM EWM items:
- Links to related issues or pull requests:

# Manual test for the reviewer
(Instructions for testing here)

# Check list for the reviewer
- [ ] best software practices
    + [ ] clearly named variables (better to be verbose in variable names)
    + [ ] code comments explaining the intent of code blocks
- [ ] All the tests are passing
- [ ] The documentation is up to date
- [ ] code comments added when explaining intent

### CODE REVIEW

File: src/webmon_app/reporting/metrics/view_util.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+            continue
+
+        if not is_acquisition_complete(run_id):
+            statuses["acquiring"] += 1
+        elif s.complete:
+            statuses["complete"] += 1
+        elif run_id.last_error() is None:
+            statuses["incomplete"] += 1
+        else:
+            statuses["error"] += 1
```
Comment (backmari) [inline src/webmon_app/reporting/metrics/view_util.py:81]: As Mat observed in the test environment, there can also be a status "unknown". In the test environment I think it is due to a developer (maybe me) adding a run straight in the database, so this shouldn't normally occur, but maybe we should include status "unknown" for completeness.

File: src/webmon_app/reporting/metrics/view_util.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
+from reporting.report.models import Instrument, DataRun, WorkflowSummary, Information
+from reporting.dasmon.models import Parameter, StatusCache, ActiveInstrument
+from reporting.report.view_util import is_acquisition_complete
+from reporting.dasmon.view_util import is_running
+from django.conf import settings
+from django.utils import timezone
+from django.db.models import Q
+
+
+def postprocessing_diagnostics():
```
Comment (backmari) [inline src/webmon_app/reporting/metrics/view_util.py:10]: can you please add docstrings to the functions in this file?

File: src/webmon_app/reporting/metrics/view_util.py (hunk 1 above)
Comment (rosswhitfield) [reply src/webmon_app/reporting/metrics/view_util.py:81]: Sure, looking at `report.view_util.get_run_status_text` I see the `unknown` status happens when you have a run but don't have a `WorkflowSummary`. I will add that here.

**Review by backmari — APPROVED** [review]
Tested locally that the status `unknown` appears in the metrics.

**Pull request merged**